RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsProducer.cc

```diff
@@ -0,0 +1,109 @@
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalRecHitCollections.h"
+#include "DataFormats/EcalRecHit/interface/EcalUncalibratedRecHit.h"
+
+#include "DataFormats/EcalDigi/interface/EcalConstants.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+
+class EcalUncalibRecHitPhase2WeightsProducer : public edm::stream::EDProducer<> {
+public:
+  explicit EcalUncalibRecHitPhase2WeightsProducer(const edm::ParameterSet& ps);
+  void produce(edm::Event& evt, const edm::EventSetup& es) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  float tRise_;
+  float tFall_;
+  std::vector<double> weights_;
+
+  edm::EDGetTokenT<EBDigiCollectionPh2> ebDigiCollectionToken_;
+  edm::EDPutTokenT<EBUncalibratedRecHitCollection> ebUncalibRecHitCollectionToken_;
+};
+
+void EcalUncalibRecHitPhase2WeightsProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<std::string>("EBhitCollection", "EcalUncalibRecHitsEB");
+  desc.add<double>("tRise", 0.2);
+  desc.add<double>("tFall", 2.);
+  desc.add<std::vector<double>>(
+      "weights",
+      {-0.121016, -0.119899, -0.120923, -0.0848959, 0.261041, 0.509881, 0.373591, 0.134899, -0.0233605, -0.0913195,
+       -0.112452, -0.118596, -0,        121737,     -0,       121737,   -0,       121737,   -0,         121737});
+  desc.add<edm::InputTag>("BarrelDigis", edm::InputTag("simEcalUnsuppressedDigis", ""));
+
+  descriptions.add("ecalUncalibRecHitPhase2Weights", desc);
```

Comment:
```suggestion
  descriptions.addWithDefaultLabel(desc);
```
since apparently this is not unique and will not be used by itself, it makes more sense to use the default, which will generate `ecalUncalibRecHitPhase2WeightsProducer_cfi`